eth/filters/filter_system.go

```diff
@@ -27,25 +28,42 @@ import (
 	"github.com/ethereum/go-ethereum/event"
 )
 
+const (
+	ChainFilter = iota
+	PendingTxFilter
+	LogFilter
+	PendingLogFilter
+)
+
 // FilterSystem manages filters that filter specific events such as
 // block, transaction and log events. The Filtering system can be used to listen
 // for specific LOG events fired by the EVM (Ethereum Virtual Machine).
 type FilterSystem struct {
 	filterMu sync.RWMutex
 	filterId int
-	filters  map[int]*Filter
-	created  map[int]time.Time
-	sub      event.Subscription
+
+	chainFilters      map[int]*Filter
+	pendingTxFilters  map[int]*Filter
+	logFilters        map[int]*Filter
+	pendingLogFilters map[int]*Filter
```

Comment:
You could still keep the generic map too to support Get, I just suggested getting rid of these individual specialized maps and merge them into a megamap :)